drivers/volume/portworx/portworx.go

```diff
@@ -820,6 +914,84 @@ func (p *portworx) VolumeDelete(pv *v1.PersistentVolume) error {
 	return p.volDriver.Delete(pv.Spec.PortworxVolume.VolumeID)
 }
 
+func (p *portworx) PerformRecovery() error {
```

Comment:
This isn't specific to portworx. Move to a common place.

Reply:
moved out of volume driver and added to rule package